File: Plugins/engines/org.eclipse.escriptmonkey.scripting.engine.python.jython.samples/samples/scripts/Editors__Replace_Selection_with_Date.py
#
# Thread:UI
# Menu: Examples >Editors > Replace Selection with Date
# Kudos: Jose Fonseca & Paul Colton & Arthur Daussy
# EnableWhen::[With selection {
#        InstanceOf "org.eclipse.jface.text.TextSelection"{
#            }
#}]::
# Description: {You can use this script to replace the current selection of the active text editor with the current date.}
# License: EPL 1.0
# 

import time 


workbench = loadModule("WorkbenchModule")
editor = workbench.getActiveEditor()

document = editor.getDocumentProvider().getDocument(editor.getEditorInput())

# get range of lines in the selection (or at the cursor position)
selection = editor.getSelectionProvider().getSelection()
offset = selection.getOffset()
deleteLength = selection.getLength()

text = time.strftime("%Y-%m-%d %H:%M:%S")
#text = "tata"
# apply edit and reveal in editor
source = document.get()
document.replace(offset, deleteLength, text)
editor.selectAndReveal(offset, len(text))

File: Plugins/engines/org.eclipse.escriptmonkey.scripting.engine.javascript.rhino.samples/samples/scripts/Editors__Replace_Selection_with_Date.js
/*
 * Thread: UI
 * Menu:   Examples > Editors > Replace Selection with Date
 * Kudos: Paul Colton (Aptana, Inc.) & Arthur Daussy
 * Description: {You can use this script to replace the current selection of the active text editor with the current date}
 * License: EPL 1.0
 * EnableWhen::[With selection {
 *        InstanceOf "org.eclipse.jface.text.TextSelection"{
 *            }
 * }]::
 */
 
function main() {
	loadModule("TextEditorModule")
	var sourceEditor = getActiveEditor();

	var valid = true;
	
	// make sure we have an editor
	if (sourceEditor === undefined) {
		valid = false;
		showError("No active editor");
	}
	
	// inject
	if (valid) {
	
			var range = sourceEditor.selectionRange;
			var offset = range.startingOffset;
			var deleteLength = range.endingOffset - range.startingOffset;
			var source = sourceEditor.source;
			
			var text = new Date();
			
			sourceEditor.applyEdit(offset, deleteLength, text);

	}
}

main()
